crates/holochain_state/src/source_chain.rs

```diff
@@ -687,43 +687,24 @@ where
                         // that has not been updated or deleted
                         txn.query_row(
                             SELECT_VALID_UNRESTRICTED_CAP_GRANT,
-                            params![CapAccess::Unrestricted.as_sql(), agent_pubkey],
+                            params![CapAccess::Unrestricted.as_sql(), author],
                             query_row_fn,
                         )
                         .optional()?
                     };
 
-                    Ok(maybe_entry)
+                    let maybe_cap_grant = maybe_entry.and_then(|entry| entry.as_cap_grant());
+                    Ok(maybe_cap_grant)
                 }
             })
-            .await?
-            .and_then(|entry| entry.as_cap_grant());
-
-        let valid_cap_grant = match maybe_cap_grant {
-            None => None,
-            Some(cap_grant) => {
-                // validate cap grant in itself, especially check if functions are authorized
-                if !cap_grant.is_valid(&check_function, &check_agent, check_secret.as_ref()) {
-                    return Ok(None);
-                }
-                match &cap_grant {
```

Comment:
Was this part of the check not being used?

Reply:
It was redundant. `CapGrant::is_valid` already performs those checks.